tests/qa_automation/xfstests_device.pm

```diff
@@ -0,0 +1,115 @@
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package xfstests_device;
+# Summary:  Device prepare related base class for xfstests_run
+# Maintainer: Yong Sun <yosun@suse.com>
+
+use strict;
+use warnings;
+use File::Basename;
+use base "opensusebasetest";
+use utils;
+use testapi qw(is_serial_terminal :DEFAULT);
+
+# Create test partition and scratch partition, and make FS in them
+sub dev_create_partition() {
+    my $self = shift;
+    assert_script_run("parted -l -m 2>&1");
+    my $cmd               = "parted -l -m 2>&1| awk -F \':\' \'{if(\$5 == \"xfs\") print \$1}\'";
+    my $test_partition_id = script_output($cmd, 10);
+    my $test_partition    = "/dev/vda" . $test_partition_id;
+
+    # seperate xfs partition(/home) into two same size xfs
+    $cmd = "parted -l -m 2>&1| awk -F \':\' \'{if(\$5 == \"xfs\") print \$2}\'";
+    my $partition_begin = script_output($cmd, 10);
+    my $extendpartition_begin = $partition_begin;
+    $cmd = "parted -l -m 2>&1| awk -F \':\' \'{if(\$5 == \"xfs\") print \$3}\'";
+    my $partition_end       = script_output($cmd, 10);
+    my $extendpartition_end = $partition_end;
+    my $partition_cut_point = $self->get_cut_point($partition_begin, $partition_end, 2);
+    assert_script_run("umount " . $test_partition);
+    type_string "parted /dev/vda\n";
```

Comment:
Not everywhere we have vda device. I think @okurz  and @mnowaksuse  know more about this ...